src/cryptography/x509.py

```diff
@@ -1158,3 +1225,18 @@ def signature_hash_algorithm(self):
         Returns a HashAlgorithm corresponding to the type of the digest signed
         in the certificate.
         """
+
+
+@six.add_metaclass(abc.ABCMeta)
+class RevokedCertificate(object):
```

Comment:
This will be part of the extension code, because that is essentially what it is. I checked with how extensions are implemented in the ABC for Certificate and it also isn't there.
So the decision would be if extensions should be a required property for the class.

I'm also planning to add little helper functions to the backend implementation (get_reason(), get_invalidity_date(), get_cert_issuer()). So this will be there, it's just not part of the required interface. But I'm fine with adding all of these to the interface specification.